packages/plugin-ext/src/plugin/tree/tree-views.ts

```diff
@@ -312,20 +315,15 @@ class TreeViewExtImpl<T> implements Disposable {
             // parent is inconsistent
             return undefined;
         }
-        const idLabel = this.getTreeItemIdLabel(treeItem);
-        let possibleIndex = children.length;
-        // find the right element id by searching all possible id names in the cache
-        while (possibleIndex-- > 0) {
-            const candidateId = this.buildTreeItemId(parentId, possibleIndex, idLabel);
-            if (this.nodes.has(candidateId)) {
-                return chain.concat(candidateId);
-            }
+        const candidateId = this.buildTreeItemId(parentId, treeItem);
+        if (this.nodes.has(candidateId)) {
+            return chain.concat(candidateId);
         }
         // couldn't calculate consistent parent chain and id
         return undefined;
     }
 
-    private getTreeItemLabel(treeItem: TreeItem): string | undefined {
+    private getLabelString(treeItem: TreeItem): string | undefined {
```

Comment:
The `String` part of this name is redundant, given the typing.

Reply:
The algorithm for computing the label is given here: https://code.visualstudio.com/api/references/vscode-api#TreeItem (label)
If there is a label string given in the item, it is used, otherwise try computing something based on the URI. The "String" is to differentiate the method from `getItemLabel`, not to indicate typing.

Reply:
If the aim is to disambiguate those two methods, then the names used should reflect the different purposes. Both return `string | undefined`, so `getLabelString` doesn't help in that regard.

Reply:
We'll have to disagree that this change makes the code clearer. If you don't mind, I'll just undo my rename of the method.